File: shapeview/src/main/java/cn/mtjsoft/www/shapeview/imp/BuilderImp.java
package cn.mtjsoft.www.shapeview.imp;

import android.graphics.drawable.GradientDrawable;

import cn.mtjsoft.www.shapeview.CustomBuilder;

/**
 * @author mtj
 * @date 2022/1/13
 * @desc
 * @email mtjsoft3@gmail.com
 */
public interface BuilderImp {

    BuilderImp setShape(@CustomBuilder.Shape int shape);

    BuilderImp setSolidColor(int solidColor);

    BuilderImp setStrokeColor(int strokeColor);

    BuilderImp setColors(int startColor, int centerColor, int endColor);

    BuilderImp setSelectColors(int startColor, int centerColor, int endColor);

    BuilderImp setOrientation(GradientDrawable.Orientation orientation);

    BuilderImp setGradientType(@CustomBuilder.GradientType int gradientType);

    BuilderImp setGradientRadius(float gradientRadius);

    BuilderImp setStrokeWidth(int strokeWidth);

    BuilderImp setDashWidth(float dashWidth);

    BuilderImp setDashGap(float dashGap);

    BuilderImp setRadius(int radius);

    BuilderImp setTopLeftRadius(int topLeftRadius);

    BuilderImp setTopRightRadius(int topRightRadius);

    BuilderImp setBottomLeftRadius(int bottomLeftRadius);

    BuilderImp setBottomRightRadius(int bottomRightRadius);

    BuilderImp setOpenSelector(boolean openSelector);

    BuilderImp setTextNormalColor(int textNormalColor);

    BuilderImp setTextSelectColor(int textSelectColor);

    BuilderImp setSolidSelectColor(int solidSelectColor);

    BuilderImp setStrokeSelectColor(int strokeSelectColor);
}

File: shapeview/src/main/java/cn/mtjsoft/www/shapeview/imp/BuilderImp.kt
package cn.mtjsoft.www.shapeview.imp

import android.graphics.drawable.GradientDrawable
import cn.mtjsoft.www.shapeview.builder.CustomBuilder
import cn.mtjsoft.www.shapeview.builder.CustomBuilder.GradientType

/**
 * @author mtj
 * @date 2022/1/13
 * @desc
 * @email mtjsoft3@gmail.com
 */
interface BuilderImp {
    fun setShape(@CustomBuilder.Shape shape: Int): BuilderImp
    fun setSolidColor(solidColor: Int): BuilderImp
    fun setStrokeColor(strokeColor: Int): BuilderImp
    fun setColors(startColor: Int, centerColor: Int, endColor: Int): BuilderImp
    fun setSelectColors(startColor: Int, centerColor: Int, endColor: Int): BuilderImp
    fun setOrientation(orientation: GradientDrawable.Orientation): BuilderImp
    fun setGradientType(@GradientType gradientType: Int): BuilderImp
    fun setGradientRadius(gradientRadius: Float): BuilderImp
    fun setStrokeWidth(strokeWidth: Int): BuilderImp
    fun setDashWidth(dashWidth: Float): BuilderImp
    fun setDashGap(dashGap: Float): BuilderImp
    fun setRadius(radius: Int): BuilderImp
    fun setTopLeftRadius(topLeftRadius: Int): BuilderImp
    fun setTopRightRadius(topRightRadius: Int): BuilderImp
    fun setBottomLeftRadius(bottomLeftRadius: Int): BuilderImp
    fun setBottomRightRadius(bottomRightRadius: Int): BuilderImp
    fun setOpenSelector(openSelector: Boolean): BuilderImp
    fun setTextNormalColor(textNormalColor: Int): BuilderImp
    fun setTextSelectColor(textSelectColor: Int): BuilderImp
    fun setSolidSelectColor(solidSelectColor: Int): BuilderImp
    fun setStrokeSelectColor(strokeSelectColor: Int): BuilderImp
}